Sources/SmileID/Classes/Helpers/LocalStorage.swift

```diff
@@ -20,169 +22,173 @@ public class LocalStorage {
-        allFiles.append(documentFront)
-        imageInfoArray.append(UploadImageInfo(imageTypeId: .idCardJpgFile, fileName: filename))
-
-        if let back = back {
-            let filename = self.filename(for: "idBack")
-            let url = try write(back, to: destinationFolder.appendingPathComponent(filename))
-            documentBack = url
-            allFiles.append(url)
-            imageInfoArray.append(
-                UploadImageInfo(imageTypeId: .idCardRearJpgFile, fileName: filename)
-            )
+            }
+            imageInfoArray.append(contentsOf: livenessImageInfos)
         }
-        let livenessInfoArray = try livenessImages?.map { [self] imageData in
-            let fileName = self.filename(for: "liveness")
-            let url = try write(imageData, to: destinationFolder.appendingPathComponent(fileName))
-            allFiles.append(url)
-            livenessImagesUrl.append(url)
-            return UploadImageInfo(imageTypeId: .livenessJpgFile, fileName: fileName)
+        if let documentFront = documentFront {
+            imageInfoArray.append(UploadImageInfo(
+                imageTypeId: .idCardJpgFile,
+                fileName: documentFront.lastPathComponent
+            ))
         }
-        if let livenessInfoArray = livenessInfoArray {
-            imageInfoArray.append(contentsOf: livenessInfoArray)
+        if let documentBack = documentBack {
+            imageInfoArray.append(UploadImageInfo(
+                imageTypeId: .idCardRearJpgFile,
+                fileName: documentBack.lastPathComponent
+            ))
         }
-        let selfieFileName = self.filename(for: "selfie")
-        let selfieUrl = try write(
-            selfie,
-            to: destinationFolder.appendingPathComponent(selfieFileName)
-        )
-        allFiles.append(selfieUrl)
-        imageInfoArray.append(
-            UploadImageInfo(imageTypeId: .selfieJpgFile, fileName: selfieFileName)
-        )
-        let idInfo = IdInfo(country: countryCode, idType: documentType)
-        let jsonData = try jsonEncoder.encode(UploadRequest(images: imageInfoArray, idInfo: idInfo))
-        let jsonUrl = try write(jsonData, to: destinationFolder.appendingPathComponent("info.json"))
-        allFiles.append(jsonUrl)
-        return DocumentCaptureResultStore(
-            allFiles: allFiles,
-            documentFront: documentFront,
-            documentBack: documentBack,
-            selfie: selfieUrl,
-            livenessImages: livenessImagesUrl
-        )
+        let data = try jsonEncoder.encode(UploadRequest(
+            images: imageInfoArray,
+            idInfo: idInfo
+        ))
+        return try createSmileFile(to: jobId, name: "info.json", file: data)
     }
 
-    private static func createDefaultDirectory() throws {
-        try createDirectory(at: defaultDirectory, overwrite: false)
+    private static func createPreUploadFile(
```

Comment:
We call this endpoint pre _p_ upload, not pre upload throughout our [rest api](https://docs.usesmileid.com/products/for-individuals-kyc/biometric-kyc#making-a-job-request-prep-upload) and [sdk docs](https://javadoc.io/doc/com.smileidentity/android-sdk/latest/com/smileidentity/models/PrepUploadResponse.html). Let's keep that consistent

```suggestion
    private static func createPrepUploadFile(
```

Reply:
Any reason we're not taking PrepUploadRequest as the parameter?

Reply:
updated